business_requirement_deliverable/models/business.py

```diff
@@ -172,6 +173,8 @@ def _compute_get_currency(self):
             currency_id = partner_id.property_product_pricelist.currency_id
             if currency_id:
                 brd.currency_id = currency_id
+            else:
+                brd.currency_id = self.env.user.company_id.currency_id.id
```

Comment:
Afaict you should remove the trailing `.id`: you want to reference the Currency record, and not it's numeric id.